File: packages/types/src/panel-settings.ts
import type { GraphPreferences } from './graph-settings';

export type PanelView = 'graph' | 'grid' | 'viewer';

export interface GridPreferences {
  // Reserved for future expansion.
}

export interface PanelPreferences {
  graph: GraphPreferences;
  grid?: GridPreferences | null;
  activeView?: PanelView | null;
  rootNodeId?: string | null;
}

File: apps/desktop/src-tauri/src/models/panel_settings.rs
use serde::{Deserialize, Serialize};

use super::graph_settings::GraphPreferences;

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct GridPreferences {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PanelPreferences {
    #[serde(default)]
    pub graph: GraphPreferences,
    #[serde(default)]
    pub grid: Option<GridPreferences>,
    #[serde(default)]
    pub active_view: Option<String>,
    #[serde(default)]
    pub root_node_id: Option<String>,
}
